weather_api.go

```diff
@@ -25,11 +24,30 @@ func (c WeatherAPIConfig) GetCurrentWeather(options Options) (*models.WeatherRes
 
 	defer resp.Body.Close()
 
-	// Check response status code
-	if resp.StatusCode != http.StatusOK {
+	// Decode JSON response body
+	var weatherResp *models.WeatherResponse
+	err = json.NewDecoder(resp.Body).Decode(&weatherResp)
+	if err != nil {
+		return nil, err
+	}
+
+	return weatherResp, nil
+}
+
+func (c WeatherAPIConfig) WeatherHistory(options Options) (*models.WeatherResponse, error) {
+	if !validateQuery(options.Query) || !validateDate(options.Date) {
```

Comment:
instead of combining two if statements, we can separate them so we can return accurate error messages.

Reply:
Good suggestion, will separate them